src/zino/events.py

```diff
@@ -147,20 +148,25 @@ def commit(self, event: Event, user: str = "monitor"):
 
         self._call_observers_for(new_event=event, old_event=old_event)
 
-    def delete(self, event: Event):
-        """Removes an event from the events dict and notifies all observers"""
-        try:
-            del self.events[event.id]
-        except KeyError:
-            pass
+    def _delete(self, event: Event):
+        """Removes a closed event from the events dict and notifies all observers"""
+        if event.state is EventState.CLOSED:
```

Comment:
I would use an early return here instead, so there would be no need for an extra indent for the rest of the method.  I.e.

```suggestion
        if event.state != EventState.CLOSED:
           return
```

Other than that, be careful to distinguish properly between `=` and `is`.  `is` is an identity test, not an equality test.  It's best to use it only when you're sure you want to match objects by identity, not by value (like `is None`, since there is always just a single `None` object in a Python runtime)